### PR TITLE
Fixes #22787 - use app logger

### PR DESCRIPTION
**Pull request by ares**
will need a rebase after #86 is merged

### CODE REVIEW

File: app/services/foreman_templates/template_importer.rb [hunk 1]
```diff
@@ -33,7 +33,9 @@ def import_from_git
       @dir = Dir.mktmpdir
 
       begin
+        logger.debug "cloned '#{@repo}' to '#{@dir}'"
         gitrepo = Git.clone(@repo, @dir)
+        logger.debug "checking out branch '#{@branch}'"
```
Comment (xprazak2) [inline app/services/foreman_templates/template_importer.rb]: Maybe `unless @branch.blank?`

**Comment by ares** [comment]
updated, please take a look

File: app/services/foreman_templates/template_importer.rb [hunk 2]
```diff
@@ -90,11 +95,19 @@ def template_model(metadata, parse_result)
...
     def save_template(template, force)
+      result = nil
       if force
-        template.ignore_locking { template.save }
+        result = template.ignore_locking { template.save }
+      else
+        result = template.save
+      end
+
+      if result?
```
[the remark this answers is not in the document]
Comment (ares) [reply app/services/foreman_templates/template_importer.rb]: yup

File: app/services/foreman_templates/template_importer.rb [hunk 3]
```diff
@@ -90,11 +95,19 @@ def template_model(metadata, parse_result)
...
+        logger.debug 'saved'
       else
-        template.save
+        logger.error "couldn't save the template because of #{template.errors.to_sentence}"
```
[the remark this answers is not in the document]
Comment (ares) [reply app/services/foreman_templates/template_importer.rb]: modified and used join with comma, and didn't feel right

**Comment by ares** [comment]
repushed, let's see if the test gets green now, they seem to run fine locally, @xprazak2 please take another look

**Comment by ares** [comment]
tests were green but rubocop failed on one style cop, disabling that as I don't see a value in this enforcement

**Comment by xprazak2** [comment]
APJ

**Comment by xprazak2** [comment]
Merging, thanks!

**Pull request merged**